### PR TITLE
add specific type to Mockito argument matcher any()

### PR DESCRIPTION
**Pull request by zhangfj88**
add specific type to Mockito argument matcher any()

### CODE REVIEW

**Comment by yaronyam** [comment]
@zhangfj88 - please let me know how you wish to progress this PR. There's a pending comment. Do you plan to handle it? If you don't have time to do so - let me know, I might handle it later. Thanks

**Comment by zhangfj88** [comment]
I agree with the solution you proposed and am currently attempting to make the modifications，Thanks

File: src/main/java/com/weirddev/testme/intellij/template/context/MockitoMockBuilder.java [hunk 1]
```diff
@@ -240,6 +241,10 @@ private String deductMatcherTypeMethod(Param param, Language language) {
         if (language != Language.Scala) {
             matcherType += "()";
         }
+        // add specific type to any()
+        if( matcherType.equals("any()")){
+            matcherType = addSpecificType(param.getType().getName());
```
[the remark this answers is not in the document]
Comment (zhangfj88) [reply src/main/java/com/weirddev/testme/intellij/template/context/MockitoMockBuilder.java:246]: I agree with the solution you proposed and am currently attempting to make the modifications，Thanks

File: src/main/java/com/weirddev/testme/intellij/template/context/MockitoMockBuilder.java [hunk 2]
```diff
@@ -226,30 +227,31 @@ public String buildMockArgsMatchers(List<Param> params,String language) {
...
+            return "anyVararg";
+        } else if(!type.isArray() && TYPE_TO_ARG_MATCHERS.containsKey(type.getCanonicalName())){
+            return TYPE_TO_ARG_MATCHERS.get(type.getCanonicalName());
+        } else {
+            return "any";
+        }
     }
 
-
+    String addSpecificType(String typeName) {
```
Comment (yaronyam) [inline src/main/java/com/weirddev/testme/intellij/template/context/MockitoMockBuilder.java:252]: eventually left unused, method can be removed, along with the import statement introducing dependency on apache commons

File: src/main/java/com/weirddev/testme/intellij/template/context/MockitoMockBuilder.java [hunk 3]
```diff
@@ -226,30 +227,31 @@ public String buildMockArgsMatchers(List<Param> params,String language) {
...
+            return matcherMethod + "("+ type.getCanonicalName()+(type.isArray()? "[]":"") +".class)";
+        } else {
+            return matcherMethod+"()";
         }
-        return matcherType;
     }
 
-    @SuppressWarnings("unused")
-    @Deprecated
-    public boolean shouldStub(Method testMethod, List<Field> testedClassFields) {
```
Comment (yaronyam) [inline src/main/java/com/weirddev/testme/intellij/template/context/MockitoMockBuilder.java:248]: please keep this method for now. Although its not used. some users might still use it in their custom templates. so it's only marked as @Deprecated for now

File: src/main/java/com/weirddev/testme/intellij/template/context/MockitoMockBuilder.java (hunk 2 above)
Comment (zhangfj88) [reply src/main/java/com/weirddev/testme/intellij/template/context/MockitoMockBuilder.java]: thanks,I will handle it.

**Comment by yaronyam** [comment]
Released in [TestMe plugin version 6.4](https://plugins.jetbrains.com/plugin/9471-testme/versions/stable/521557)

**Pull request merged**